src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/CharSetSolver.cs

```diff
@@ -36,7 +36,7 @@ internal sealed class CharSetSolver : ISolver<BDD>
         /// This cache is necessary for the recursive operation algorithms to be guaranteed linear time.
         /// A well-crafted character class could otherwise cause execution time to be exponential.
         /// </remarks>
-        private readonly Dictionary<(BooleanOperation op, BDD a, BDD? b), BDD> _operationCache = new();
+        private readonly Dictionary<(int op, BDD a, BDD? b), BDD> _operationCache = new(); // op is BooleanOperation; using int to reuse generic instantiation
```

Comment:
(nit)

```suggestion
        private readonly Dictionary<(int op, BDD a, BDD? b), BDD> _operationCache = new(); // op is BooleanOperation; using int to reuse generic instantiation with _bddCache above
```

Might help explain a little bit more.